From ee7e02737b599d90f628252202e5a9969d050f5e Mon Sep 17 00:00:00 2001
From: Lukas Bulwahn <lukas.bulwahn@oss.bmw-carit.de>
Date: Sat, 1 Mar 2014 16:32:32 +0100
Subject: [PATCH] correct path for CONFIGURE_FILE in CMakeLists.txt

If CMAKE_MODULE_PATH is empty before appending
'${CMAKE_CURRENT_SOURCE_DIR}/cmake' to it (CMakeLists.txt:27),
CONFIGURE_FILE (CMakeLists.txt:182) works as intended.
However, when CMAKE_MODULE_PATH is not empty, e.g., because of
further tool-chain settings, CONFIGURE_FILE fails with:

CMake Error: File <SOME-OTHER-PATH>;<POCO-SRC-PATH>/cmake/cmake_uninstall.cmake.in does not exist.
| CMake Error at CMakeLists.txt:182 (CONFIGURE_FILE):
|   configure_file Problem configuring file

This commit addresses this problem by only using
'${CMAKE_CURRENT_SOURCE_DIR}/cmake' in CONFIGURE_FILE.
The problem was discovered in the meta-ros layer project
(http://github.com/bmwcarit/meta-ros) when creating a libpoco
bitbake recipe, which uses poco's cmake installation scripts.

Signed-off-by: Lukas Bulwahn <lukas.bulwahn@oss.bmw-carit.de>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8d02f13..1c9f523 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -180,7 +180,7 @@ endif(APRUTIL_FOUND AND APACHE_FOUND)
 #############################################################
 # Uninstall stuff see: http://www.vtk.org/Wiki/CMake_FAQ
 CONFIGURE_FILE(
-  "${CMAKE_MODULE_PATH}/cmake_uninstall.cmake.in"
+  "${CMAKE_CURRENT_SOURCE_DIR}/cmake/cmake_uninstall.cmake.in"
   "${CMAKE_CURRENT_BINARY_DIR}/cmake_uninstall.cmake"
   IMMEDIATE @ONLY)
 
-- 
1.8.3.1

